### ISSUE TITLE
Route accelerator microbenchmarks away from fresh-process pyperf timing

### ISSUE THREAD

**Issue by morluto** [morluto/flameox#371]
## Problem

Flameox exposes both `pyperf` and producer-neutral `benchmark-samples`, but capture planning does not steer accelerator microbenchmarks toward the latter. A CUDA workload can therefore spend minutes producing statistically tidy samples of interpreter, framework, and CUDA-context startup while obscuring the operation being optimized.

The pyperf adapter is behaving as a process benchmark. The gap is admission/routing: its current warning mentions experiment randomization but does not explain that every sample launches a fresh workload process or suggest the in-process structured adapter.

## Reproduction

I benchmarked a real Slime GAE workload on an RTX 3060 with Torch 2.13.0. The command creates CUDA tensors shaped `[64, 8192]`, warms and runs both `vanilla_gae` and `chunked_gae`, and synchronizes the device.

### pyperf capture

```console
$ flameox capture run pyperf --workload slime-gae-gpu-benchmark --workspace <workspace> --json
```

The adapter used three pyperf worker processes, three values, and one warmup. Observed evidence:

- capture wall time: 122.51 s
- nine measured command samples: 7.34–7.80 s each
- seven warmup samples
- every sample had `loop_count=1`

These are valid fresh-process command timings, but they are dominated by Python, Torch import, CUDA initialization, tensor allocation, and setup. They cannot resolve the operation-level difference.

### Structured in-process capture

The same workload was adapted to emit `flameox.benchmark-samples.v1` with exact raw samples, explicit warmups, `measurement_clock=host_monotonic`, `synchronization=device_synchronize`, CUDA device identity, and shape/dtype dimensions:

```console
$ flameox capture run benchmark-samples --workload slime-gae-gpu-benchmark --workspace <workspace> --json
$ flameox extract benchmark-samples <run-id> --workspace <workspace> --json
```

Observed steady-state samples:

- `gae.vanilla`: 519–551 ms
- `gae.chunked`: 3.43–3.49 ms
- first chunked warmup: 84.15 ms
- total capture wall time: 8.17 s

The structured route preserved ten measured samples and four warmups with the correct synchronization semantics. It made the roughly 150× improvement visible and showed why the first warmup must not be mixed into steady-state evidence.

PyTorch also provides the maintained `torch.utils.benchmark.Timer`, which is Torch-aware, handles CUDA synchronization, and supports blocked autorange for operation-level measurement:
https://github.com/pytorch/pytorch/blob/main/docs/source/benchmark_utils.md

## Proposed outcome

Make benchmark planning distinguish process/startup benchmarks from operation or accelerator benchmarks and route accordingly.

- Describe `pyperf` as fresh-process command timing in plan semantics and warnings.
- When workload identity requires CUDA/HIP/XPU or the investigation asks about an operator/kernel, recommend `benchmark-samples` rather than pyperf.
- Generate a concrete structured-sample harness contract, including clock, synchronization, warmup, scope, device, loop count, and dimensions.
- Consider an optional maintained PyTorch benchmark producer based on `torch.utils.benchmark.Timer`, with the raw samples normalized through the existing producer-neutral schema.
- Keep pyperf available for startup, CLI, and true end-to-end process workloads.

## Success criteria

- A plan makes clear whether one sample is a process invocation or an in-process operation.
- Accelerator workloads receive a typed recommendation for synchronized raw samples.
- The recommendation includes a runnable next action or minimal schema template, not only prose.
- Warmups and steady-state values remain distinct through capture, extraction, comparison, and reporting.
- Tests demonstrate that a microsecond/millisecond CUDA operation is not represented as a multi-second pyperf operation without an explicit process-scope choice.
- Documentation compares pyperf, `benchmark-samples`, and any Torch-aware producer by intended scope.

**Comment by morluto** [comment]
Implemented on main in 73fde34. pyperf is now explicitly modeled as fresh-process command timing, while accelerator and operator investigations are routed toward structured in-process samples. The new torch.benchmark path uses torch.utils.benchmark.Timer after setup and warmup, keeps raw warmup and measured samples distinct, records synchronization and dimensions, bounds sample count and duration, and records CUDA-event time only as a separate metric. Planning and documentation now make the estimand and next action explicit.
